### PR TITLE
Cmake fixes and upgrades for glslang, sdk locations

### LINKED ISSUE #113: CMake issues building on Windows + VS 2015
I had some struggles getting this repository to build since separating the repos and I figured it was worth sharing the problems since some of them may be worth fixing.

1. As far as I can figure out, it's not possible to use cmake-gui any more since command-line options are required to set GLSLANG_REPO_ROOT and I can't see any way to set it in the GUI since it's not a variable that's exposed.  Attempting to configure in cmake-gui I get:

ERRORMust define GLSLANG_REPO_ROOT -- see BUILD.md

I wouldn't be surprised if this impacts ccmake on Linux as well? (haven't tried)  This used to work in the old combined repo.

2. As far as I can find in BUILD.md, there was nothing to tell me that I needed to build glslang in-tree with folder name build (64-bit) or build32 (32-bit).  This should be documented and/or configurable.

3. I happened to build glslang RelWithDebInfo and the libraries were not found.  Digging through CMakeLists.txt I discovered only Release/Debug builds will be found.  This should also probably be documented.

4. I had to explicitly set LIBVK in addition to LOADER_REPO_ROOT.  Is that expected?  It wasn't documented.  This is the cmake command-line I'm using:

```
C:\Users\dang\src\Vulkan-ValidationLayers\build>cmake -DLOADER_REPO_ROOT=C:\VulkanSDK\1.1.73.0 -DGLSLANG_REPO_ROOT=c:\users\dang\src\glslang -DLIBVK=c:\VulkanSDK\1.1.73.0\Lib\vulkan-1.lib -G "Visual Studio 14 2015 Win64" ..
```

### PR DESCRIPTION
**Pull request by mark-lunarg**
Hopefully addresses three issues found by @danginsburg.  

- If an SDK is installed, setting LOADER_REPO_ROOT is not required.  However, setting it to point to an installed SDK uncovered a cmake deficiency.
- Pointing to a glslang repo caused a host of issues (build dir names, build decorations, etc).  Changed glslang location method to **require** building per the glslang instructions, meaning that  INSTALL_PREFIX is supplied and 'make install' gets run.  GLSLANG_REPO_ROOT will be phased out and the new hotness is "**GLSLANG_INSTALL_DIR**".  Both should work for the time being.
- The LOADER and GLSLSANG variables were not predefined, and so could not be manipulated with ccmake or cmake-gui.  

Updated BUILD.md to reflect changes.

Fixes #113.

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -40,7 +40,8 @@ Windows 7+ with the following software packages:
 
 - Microsoft Visual Studio 2013 Update 4 Professional, VS2015 (any version), or VS2017 (any version).
 - [glslang](https://github.com/KhronosGroup/glslang)
-  - Ensure that the 'update_glslang_sources.py' script has been run, and the repository successfully built.
+  - Ensure that the 'update_glslang_sources.py' script has been run. Follow the build instructions in the
+    glslang README.md file, noting the location of the chosen install directory.
```
Comment (tobine) [inline BUILD.md]: Make README.md a click-able link:

`[README.md](https://github.com/KhronosGroup/glslang/blob/master/README.md)`

File: BUILD.md [hunk 2]
```diff
@@ -139,9 +140,13 @@ It should be straightforward to adapt this repository to other Linux distributio
 Vulkan Loader Library
   - Building the Layer Validation Tests requires linking to the Vulkan Loader Library (libvulkan-1.so).
       - The LOADER_REPO_ROOT environment variable should be used on the cmake command line to specify a vulkan loader library. Make sure to specify an absoute path, like so:
-      
+
           `cmake -DLOADER_REPO_ROOT=c:\development\Vulkan-Loader ....`
 
+- [glslang](https://github.com/KhronosGroup/glslang)
+  - Ensure that the 'update_glslang_sources.py' script has been run. Follow the build instructions in the
+    glslang README.md file, noting the location of the chosen install directory.
```
Comment (tobine) [inline BUILD.md]: Can link README.md here as well

File: BUILD.md [hunk 3]
```diff
@@ -76,9 +77,9 @@ Windows 7+ with the following software packages:
 
 For example, for VS2017 (generators for other versions are [specified here](#cmake-visual-studio-generators)):
 
-    cmake -DLOADER_REPO_ROOT=c:\absolute_path_to\VULKAN_LOADER -DGLSLANG_REPO_ROOT=c:\absolute_path_to\glslang -G "Visual Studio 15 2017 Win64" ..
+    cmake -DLOADER_REPO_ROOT=c:\absolute_path_to\VULKAN_LOADER -DGLSLANG_INSTALL_DIR=c:\absolute_path_to\glslang\install -G "Visual Studio 15 2017 Win64" ..
 
-Both LOADER_REPO_ROOT and GLSLANG_REPO_ROOT each require absolute paths.
+Both LOADER_REPO_ROOT and GLSLANG_INSTALL_DIR each require absolute paths.
```
Comment (jzulauf-lunarg) [inline BUILD.md]: Are we going to convert LOADER_REPO_ROOT to also be LOADER_INSTALL_DIR?

In the code we say that GLSLANG_REPO_ROOT is deprecated... probably should say so in the text, yes?

File: CMakeLists.txt [hunk 4]
```diff
@@ -146,71 +146,97 @@ endif()
 option(BUILD_TESTS "Build tests" ON)
 option(BUILD_LAYERS "Build layers" ON)
 
+set(NOT_DEFINED "NOT_DEFINED")
```
Comment (jzulauf-lunarg) [inline CMakeLists.txt]: No other way to test for undefined/empty string?

Comment (jzulauf-lunarg) [reply CMakeLists.txt]: "if(DEFINED <variable>)" from https://cmake.org/cmake/help/v3.0/command/if.html  -- but I don't know if this is best practice or not...

File: BUILD.md (hunk 3 above)
Comment (mark-lunarg) [reply BUILD.md]: We'll probably move to this model for all repos, someday.

**Comment by mark-lunarg** [comment]
Addressed the fine suggestions in a couple of new commits.

**Pull request merged**